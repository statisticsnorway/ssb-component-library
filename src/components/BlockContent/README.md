BlockContent
========

> A page layout component to wrap around lots of content 

### Usage

#### HTML

```html
<div class="ssb-block-content">
    <div class="section-number">1</div>
    Insert content here
</div>
```

#### React

```jsx harmony
<BlockContent sectionNumber={1}>
  Insert content here
</BlockContent> 
```

Available props:

| Name       | Type           | Description  |
| ---------- | ------------- | ----- |
| children   | Required node | Content within the wrapper |
| sectionNumber | Optional number or string | Optional rendering of a page number |
