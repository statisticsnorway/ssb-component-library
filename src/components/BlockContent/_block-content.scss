.ssb-block-content {
  @include open-sans;
  background: $ssb-white;
  border-top: 4px solid $ssb-green-3;
  box-shadow: 0 2px 4px 1px rgba(200, 200, 200, .5);
  box-sizing: border-box;
  max-width: 1200px;
  padding: 100px;
  position: relative;
  width: 100%;

  .page-number {
    @include roboto;
    color: $ssb-dark-2;
    font-size: 30px;
    margin-top: -35px;

    &:before {
      background: $ssb-green-3;
      content: '';
      display: inline-block;
      height: 22px;
      margin-right: 10px;
      width: 10px;
    }
  }
}
