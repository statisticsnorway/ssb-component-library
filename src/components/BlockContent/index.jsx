import React from 'react';
import PropTypes from 'prop-types';

const BlockContent = ({children, pageNumber}) => (
	<div className="ssb-block-content">
		{pageNumber && <div className="page-number">{pageNumber}</div>}
		{children}
	</div>
);

BlockContent.propTypes = {
	children: PropTypes.node.isRequired,
	pageNumber: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
};

export default BlockContent;
