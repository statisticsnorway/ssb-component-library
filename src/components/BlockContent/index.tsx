import React, { type ReactNode } from 'react'

interface BlockContentProps {
  children: ReactNode
  className?: string
  sectionNumber?: number | string
}

const BlockContent: React.FC<BlockContentProps> = ({ children, className, sectionNumber }) => (
  <div className={`ssb-block-content${className ? ` ${className}` : ''}`}>
    {sectionNumber && (
      <div className='section-number'>
        {(typeof sectionNumber === 'string' && sectionNumber.length === 1) ||
        (typeof sectionNumber === 'number' && sectionNumber < 10)
          ? '0'
          : ''}
        {sectionNumber}
      </div>
    )}
    {children}
  </div>
)

export default BlockContent
