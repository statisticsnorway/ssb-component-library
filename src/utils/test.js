import fs from 'fs';
import path from 'path';
import { render as rtlRender } from '@testing-library/react';

export * from '@testing-library/react';
export * from '@testing-library/jest-dom';

const styles = fs.readFileSync(path.relative(process.cwd(), './lib/bundle.css'), 'utf8');

export function render(ui, options) {
	const view = rtlRender(ui, options);

	const styleElement = document.createElement('style');
	styleElement.innerHTML = styles;
	document.body.appendChild(styleElement);

	return view;
}
