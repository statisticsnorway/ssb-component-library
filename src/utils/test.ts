import fs from 'fs'
import path from 'path'
import { RenderOptions, RenderResult, render as rtlRender } from '@testing-library/react'
import { ReactNode } from 'react'

export * from '@testing-library/react'

const styles = fs.readFileSync(path.relative(process.cwd(), './lib/bundle.css'), 'utf8')

export function render(ui: ReactNode, options?: RenderOptions): RenderResult {
  const view = rtlRender(ui, options)

  const styleElement = document.createElement('style')
  styleElement.innerHTML = styles
  document.body.appendChild(styleElement)

  return view
}
