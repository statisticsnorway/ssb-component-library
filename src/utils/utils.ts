// utils.ts
const classNames = (...classes: (string | boolean | undefined)[]): string => classes.filter(Boolean).join(' ')

export default classNames
