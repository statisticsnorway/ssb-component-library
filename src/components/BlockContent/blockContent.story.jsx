import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered/react';
import BlockContent from './index';
import Title from '../Title';
import LeadParagraph from '../LeadParagraph';

storiesOf('BlockContent', module).addDecorator(centered)
	.add('Default', () => (
		<div style={{ width: '900px' }}>
			<BlockContent sectionNumber={1}>
				<Title noMargin size={1}>Overskrift</Title>
				<LeadParagraph>Her er det sÃ¥ mye innhold at man bÃ¸r innlede</LeadParagraph>
			</BlockContent>
		</div>
	));
