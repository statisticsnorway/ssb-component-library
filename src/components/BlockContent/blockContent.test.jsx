import React from 'react'
import { screen, render } from '../../utils/test'
import BlockContent from './index'

describe('BlockContent component', () => {
  test('Matches the snapshot', () => {
    const { asFragment } = render(<BlockContent>BlockContent</BlockContent>)
    expect(asFragment()).toMatchSnapshot()
  })
  test('Add 0 infront of sectionnumbers under 10', async () => {
    render(<BlockContent sectionNumber={5}>BlockContent</BlockContent>)
    expect(await screen.findByText('05')).toBeVisible()
  })
  test('Not add 0 infront of sectionnumbers above 9', async () => {
    render(<BlockContent sectionNumber={22}>BlockContent</BlockContent>)
    expect(await screen.queryByText('022')).toBeNull()
    expect(await screen.findByText('22')).toBeVisible()
  })
})
