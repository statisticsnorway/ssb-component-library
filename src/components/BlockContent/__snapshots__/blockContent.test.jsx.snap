// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockContent component Matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="ssb-block-content"
  >
    BlockContent
  </div>
</DocumentFragment>
`;
